tests/integration/test_charm.py

```diff
@@ -231,9 +230,35 @@ async def test_seldon_alert_rules(ops_test: OpsTest):
     assert seldon_reconcile_error_alert["state"] == "firing"
 
     # cleanup SeldonDeployment
-    client.delete(seldon_deployment, name="seldon-model-1", namespace=namespace)
+    client.delete(seldon_deployment, name="seldon-model-1", namespace=namespace, grace_period=0)
+
+    # wait for application to settle
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME], status="active", raise_on_blocked=True, timeout=60, idle_period=30
+    )
+
+
+def create_seldon_deployment():
+    """Create Seldon Deployment once.
+
+    If does not exist, create Seldon Deployment and store it in global.
+    Return Seldon Deployment.
+    """
+    global SELDON_DEPLOYMENT
+    if SELDON_DEPLOYMENT is None:
+        seldon_deployment = create_namespaced_resource(
+            group="machinelearning.seldon.io",
+            version="v1",
+            kind="seldondeployment",
+            plural="seldondeployments",
+            verbs=None,
+        )
+        SELDON_DEPLOYMENT = seldon_deployment
+
+    return SELDON_DEPLOYMENT
```

Comment:
As an alternative, you can skip the function and `global` by including this at the top of this file instead.  See [this comment]().  Then instead of doing
```python
seldon_deployment = create_seldon_deployment()
sdep = seldon_deployment(yaml.safe_load(...))
```
in the tests, you can do:
```python
sdep = SELDON_DEPLOYMENT(yaml.safe_load(...))
```

Reply:
Great idea! Will do.